osu.Game.Rulesets.Osu/OsuDifficulty/Preprocessing/OsuDifficultyHitObject.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using osu.Game.Rulesets.Osu.Objects;
+
+namespace osu.Game.Rulesets.Osu.OsuDifficulty.Preprocessing
+{
+    public class OsuDifficultyHitObject
+    {
+        public OsuHitObject BaseObject { get; }
+
+        /// <summary>
+        /// Normalized distance from the StartPosition of the previous note.
+        /// </summary>
+        public double Distance { get; private set; }
+
+        /// <summary>
+        /// Milliseconds elapsed since the StartTime of the previous note.
+        /// </summary>
+        public double Ms { get; private set; }

```

Comment:
Call this `DeltaMs` to make clear, that it's a _difference_ rather than the absolute timestamp. In fact, I'd prefer if you called it `DeltaTime` to remain consistent with the rest of the project. The units can still be described in the xml-doc.